networkx/algorithms/shortest_paths/generic.py

```diff
@@ -221,22 +221,24 @@ def shortest_path_length(G, source=None, target=None, weight=None):
         if target is None:
             ## Find paths between all pairs.
             if weight is None:
-                paths=nx.all_pairs_shortest_path_length(G)
+                paths = nx.all_pairs_shortest_path_length(G)
             else:
                 paths=nx.all_pairs_dijkstra_path_length(G, weight=weight)
         else:
             ## Find paths from all nodes co-accessible to the target.
             with nx.utils.reversed(G):
                 if weight is None:
-                    paths=nx.single_source_shortest_path_length(G, target)
+                    # TODO, for some reason this is not working.
+                    # need to investigate it further.
+                    paths = dict(nx.single_source_shortest_path_length(G, target))
```

Comment:
For some reason this is not working. Lot of errors pop up in the connectivity module, everything else is working fine :/